### PR TITLE
Bigtable: refactor update_app_profile() to remove update_mask argument

### PR DESCRIPTION
**Pull request by AVaksman**
refactored update_app_profile() to remove update_mask argument
added asserts to unit tests
added system tests

### CODE REVIEW

**Comment by sduskis** [comment]
@AVaksman, there's a coverage issue:

> google/cloud/bigtable/instance.py          110      0     32      1    99%   480->483

**Comment by AVaksman** [comment]
@sduskis thanks.
Fixed unit tests coverage

File: bigtable/google/cloud/bigtable/enums.py [hunk 1]
```diff
@@ -53,3 +53,27 @@ class InstanceType(object):
...
+
+      SINGLE (int): Unconditionally routes all read/write requests to a
+      specific cluster.
+      This option preserves read-your-writes consistency, but does not improve
+      availability.
+      See
+      https://cloud.google.com/bigtable/docs/reference/admin/rpc/google.bigtable.admin.v2#google.bigtable.admin.v2.AppProfile.SingleClusterRouting
+    """
+    ANY = 1
+    SINGLE = 2
```
Comment (tseaver) [inline bigtable/google/cloud/bigtable/enums.py]: It seems odd to have bare `int` literals here, when the rest of the module points to the protobuf-generated code.  Should this enum be present in `google.cloud.bigtable_admin_v2.enums`, and aliased here?

File: bigtable/tests/system.py [hunk 2]
```diff
@@ -210,33 +209,118 @@ def test_update(self):
...
+        self.assertEqual(app_profile, alt_app_profile)
+
+        # Modify existing app_profile to singly routing policy and confirm
+        new_description = 'To single routing policy'
+        allow_transactional_writes = False
+        operation = instance.update_app_profile(
+            app_profile_id=app_profile_id,
+            routing_policy_type=ROUTING_POLICY_TYPE_SINGLE,
+            description=new_description,
+            ignore_warnings=ignore_warnings,
```
Comment (tseaver) [inline bigtable/tests/system.py]: For completeness, should we be updating `ignore_warnings` to False here?

File: bigtable/tests/system.py [hunk 3]
```diff
@@ -210,33 +209,118 @@ def test_update(self):
...
+            alt_app_profile.single_cluster_routing.allow_transactional_writes,
+            allow_transactional_writes)
+
+        # Modify existing app_proflie to multi cluster routing
+        new_description = 'To multi cluster routing'
+        operation = instance.update_app_profile(
+            app_profile_id=app_profile_id,
+            routing_policy_type=ROUTING_POLICY_TYPE_ANY,
+            description=new_description,
+            ignore_warnings=ignore_warnings)
```
Comment (tseaver) [inline bigtable/tests/system.py]: Likewise here?

File: bigtable/tests/system.py (hunk 2 above)
Comment (sduskis) [reply bigtable/tests/system.py]: In general, it's a good idea to `ignore_warnings=True` in code.  `ignore_warnings=False` only seems to make sense for UI purposes.

Comment (tseaver) [reply bigtable/tests/system.py]: OK.

**Pull request merged**